app/generic/verifier/facades/acapy_verifier_utils.py

```diff
@@ -0,0 +1,73 @@
+from typing import Optional, Union
+
+from aries_cloudcontroller import V10PresentationExchange, V20PresExRecord
+
+from app.generic.verifier.models import (
+    PresentationExchange,
+    ProofRequestProtocolVersion,
+)
+
+
+def pres_id_no_version(proof_id: str) -> str:
+    if proof_id.startswith("v2-") or proof_id.startswith("v1-"):
+        return proof_id[3:]
+    else:
+        raise ValueError("proof_id must start with prefix v1- or v2-")
+
+
+def record_to_model(
+    record: Union[V20PresExRecord, V10PresentationExchange]
+) -> PresentationExchange:
+    if isinstance(record, V20PresExRecord):
+        return PresentationExchange(
+            connection_id=record.connection_id,
+            created_at=record.created_at,
+            protocol_version=ProofRequestProtocolVersion.v20.value,
+            presentation=record.pres,
+            proof_id="v2-" + str(record.pres_ex_id),
+            role=record.role,
+            state=record.state,
+            updated_at=record.updated_at,
+            verified=string_to_bool(record.verified),
+        )
+    elif isinstance(record, V10PresentationExchange):
+        return PresentationExchange(
+            connection_id=record.connection_id,
+            created_at=record.created_at,
+            presentation=record.presentation,
+            protocol_version=ProofRequestProtocolVersion.v10.value,
+            proof_id="v1-" + str(record.presentation_exchange_id),
+            role=record.role,
+            state=state_to_rfc_state(record.state),
+            updated_at=record.updated_at,
+            verified=string_to_bool(record.verified),
+        )
+    else:
+        raise ValueError("Record format unknown.")
+
+
+def string_to_bool(verified: Optional[str]) -> Optional[bool]:
+    if verified == "true":
+        return True
+    elif verified == "false":
+        return False
+    else:
+        return None
+
+
+def state_to_rfc_state(state: Optional[str]) -> Optional[str]:
+    translation_dict = {
+        "proposal_sent": "proposal-sent",
+        "proposal_received": "proposal-received",
+        "request_sent": "request-sent",
+        "request_received": "request-received",
+        "presentation_sent": "presentation-sent",
+        "presentation_received": "presentation-received",
+        "done": "done",
+        "abandoned": "abandoned",
+    }
```

Comment:
Is the start state the same for both v1 and v2?

Reply:
Can quickly discuss this also in the call for unifying the presentation. Not 100% sure I know what you mean.

Reply:
This is only used for v1. v2 has the `-` instead of the `_` already.